From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jolla.com>
Date: Wed, 17 Nov 2021 14:13:00 +0200
Subject: [PATCH] disabled gtk-doc

---
 autogen.sh   | 11 +++++++----
 configure.ac |  9 +++++----
 2 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 4d4a8b8b..a2f16071 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -41,12 +41,15 @@ if $have_libtool ; then : ; else
 	DIE=1
 fi
 
-(gtkdocize --version) < /dev/null > /dev/null 2>&1 || {
+gtkdocize=gtkdocize
+($gtkdocize --version) < /dev/null > /dev/null 2>&1 || {
 	echo
-	echo "You must have gtk-doc installed to compile $PROJECT."
+	echo "You must have gtk-doc installed to compile docs for $PROJECT."
 	echo "Install the appropriate package for your distribution,"
 	echo "or get the source tarball at ftp://ftp.gnome.org/pub/GNOME/sources/gtk-doc/"
-	DIE=1
+	#DIE=1
+	echo "EXTRA_DIST = missing-gtk-doc" > gtk-doc.make
+	gtkdocize=true
 }
 
 (autoconf --version) < /dev/null > /dev/null 2>&1 || {
@@ -94,7 +97,7 @@ touch ChangeLog
 ${ACLOCAL} -I m4/ ${ACLOCAL_FLAGS} || exit $?
 
 ${LIBTOOLIZE} --force || exit $?
-gtkdocize || exit $?
+$gtkdocize
 
 autoheader || exit $?
 
diff --git a/configure.ac b/configure.ac
index c6c60296..f2cdad40 100644
--- a/configure.ac
+++ b/configure.ac
@@ -187,16 +187,17 @@ AC_CHECK_FUNCS(select poll)
 dnl ************************************
 dnl Checks for gtk-doc and docbook-tools
 dnl ************************************
+dnl NOTE: We need to use a separate automake conditional for this
+dnl       to make this work with the tarballs.
+AM_CONDITIONAL(ENABLE_GTK_DOC, test "x$enable_gtk_doc" = "xyes")
 
+m4_ifdef([GTK_DOC_CHECK], [
 GTK_DOC_CHECK([1.8])
+])
 
 AC_PATH_PROGS([DB2HTML], [db2html docbook2html])
 AM_CONDITIONAL(HAVE_DOCBOOK, [test -n "$DB2HTML"], "true", "false")
 
-dnl NOTE: We need to use a separate automake conditional for this
-dnl       to make this work with the tarballs.
-AM_CONDITIONAL(ENABLE_GTK_DOC, test "x$enable_gtk_doc" = "xyes")
-
 dnl Check for profiling options
 AC_ARG_ENABLE([profiling],
               AS_HELP_STRING([--enable-profiling],[enable profiling compile flags [[default=no]]]),,
