From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeffrey Stedfast <jestedfa@microsoft.com>
Date: Sat, 11 May 2024 11:43:36 -0400
Subject: [PATCH] Check ac_cv_sys_file_offset_bits against empty string

May fix issue #158
---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f2cdad40..20994707 100644
--- a/configure.ac
+++ b/configure.ac
@@ -154,6 +154,7 @@ AC_CHECK_HEADERS(sys/param.h)
 AC_CHECK_HEADERS(winsock2.h)
 AC_CHECK_HEADERS(string.h)
 AC_CHECK_HEADERS(stdlib.h)
+AC_CHECK_HEADERS(stdint.h)
 AC_CHECK_HEADERS(netdb.h)
 AC_CHECK_HEADERS(time.h)
 AC_CHECK_HEADERS(poll.h)
@@ -432,7 +433,7 @@ if test "x$enable_largefile" != "xno"; then
         enable_largefile="yes"
     fi
     
-    if test "x$ac_cv_sys_file_offset_bits" != "xno"; then
+    if test "x$ac_cv_sys_file_offset_bits" != "x"; then
         LFS_CFLAGS="$LFS_CFLAGS -D_FILE_OFFSET_BITS=$ac_cv_sys_file_offset_bits"
         enable_largefile="yes"
     fi
